docs/howtos/DataPlane Generation - DPG/08methodInputs.mdx

```diff
@@ -0,0 +1,513 @@
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+alias User = {
+  @path id: string;
+  firstName: string;
+  lastName: string;
+};
+
+op upload(...User): void;
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+```python
+def upload(id: str, first_name: str, last_name: str) -> None:
+  ...
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+public void upload(String id, String firstName, String lastName);
+```
+
+</TabItem>
+<TabItem value="go" label="Go" >
+
+```go
+
+```
+
+</TabItem>
+</Tabs>
+
+### Named model
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+model User {
+  firstName: string;
+  lastName: string;
+}
+
+op upload(...User): void;
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+```python
+class User:
+  first_name: str
+  last_name: str
+
+def upload(user: User) -> None:
+  ...
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+// Model class
+@Immutable
+public final class User implements JsonSerializable<User> {
+    public User(String firstName, String lastName);
+    public String getFirstName();
+    public String getLastName();
+}
+
+// Client API
+public void upload(User user);
+```
+
+</TabItem>
+<TabItem value="go" label="Go" >
+
+```go
+
+```
+
+</TabItem>
+</Tabs>
+
+### Model with @body Property
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+model User {
+  firstName: string;
+  lastName: string;
+}
+
+model UserRequest {
+  @body user: User;
+}
+
+op upload(...UserRequest): void;
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+For Python, we will also generate the overloads described in the Http Post section, but omitting for brevity
+
+```python
+class User:
+  first_name: str
+  last_name: str
+
+class UserRequest:
+  user: User
+
+def upload(userRequest: UserRequest) -> None:
+  ...
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+// Model class
+@Immutable
+public final class User implements JsonSerializable<User> {
+    public User(String firstName, String lastName);
+    public String getFirstName();
+    public String getLastName();
+}
+
+// Client API
+public void upload(User user);
+```
+
+</TabItem>
+<TabItem value="go" label="Go" >
+
+```go
+
+```
+
+</TabItem>
+</Tabs>
+
+### Model with Decorated Properties
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+model BlobProperties {
+  @key
+  @visibility("read")
```

Comment:
Can `@visibility("read")` decorate a path parameter together with `@key` decorator? 
```suggestion
```

Reply:
It is permitted: 
![image](https://github.com/Azure/typespec-azure/assets/70930885/e936e1ac-590a-48bb-801b-a08d40a5723b)

Reply:
nit: it seems `@path` is not necessary if it is already decorated with `@key`